x/go/binary/codec.go

```diff
@@ -0,0 +1,248 @@
+package binary
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"github.com/synnaxlabs/alamos"
+	"github.com/synnaxlabs/x/errors"
+	"github.com/vmihailenco/msgpack/v5"
+	"io"
+	"reflect"
+	"strconv"
+)
+
+// sugarEncodingErr adds additional context to encoding errors.
+func sugarEncodingErr(_ context.Context, value interface{}, err error) error {
```

Comment:
Why do these two methods need to take in the context argument if it doesnt use it? In the previous implementation, encode/decode was never used in any codec methods